mlrun/projects/project.py

```diff
@@ -2437,9 +2437,21 @@ def _instantiate_function(
             )
 
         if isinstance(func, str):
-            # in hub or db functions name defaults to the function name
-            if not name and not (func.startswith("db://") or func.startswith("hub://")):
-                raise ValueError("Function name must be specified")
+            # In hub or db functions name defaults to the function name
+            # If name is specified, we try to create a new function with this name.
+            if not name:
+                if func and not (func.startswith("db://") or func.startswith("hub://")):
+                    existing_func_names = []
+                    existing_func_defs = self.list_functions(name=func)
+                    if existing_func_defs:
+                        existing_func_names = [
+                            existing_func_def.metadata.name
+                            for existing_func_def in existing_func_defs
+                        ]
+                    if not existing_func_defs or func not in existing_func_names:
+                        raise mlrun.MLRunInvalidArgumentError(
+                            f'Function "{func}" does not exist in the project "{self.name}".'
+                        )
```

Comment:
Since we are only using this `existing_func_names` to check the current func anyway, I'd probably have the if already in the comprehension. Also the if existing_func_defs is a bit redundant and causes the if to be a bit complicated. How about?
```suggestion
                    existing_func_names = [
                        existing_func_def.metadata.name
                        for existing_func_def in self.list_functions(name=func)
                        if func == existing_func_def.metadata.name
                    ]
                    # expect list with length 1
                    if not existing_func_names:
                        raise mlrun.MLRunInvalidArgumentError(
                            f'Function "{func}" does not exist in the project "{self.name}".'
                        )
```

Reply:
This was committed by accident and was removed 😅